lib/ecto/schema.ex

```diff
@@ -2385,4 +2386,13 @@ defmodule Ecto.Schema do
       opts
     end
   end
+
+  defp expand_nested_module_alias({:__aliases__, _, [h | t]}, env) when is_atom(h) do
+    module = Module.concat([env.module, h])
+
+    case t do
+      [] -> module
+      _ -> String.to_atom(Enum.join([module | t], "."))
+    end
```

Comment:
I think you can write it as `Module.concat([env.module, h | t])`?

You also need to handle two cases:

1. `embeds_many :foo, Elixir.MyModule` - In this case, just return the alias is, as Elixir means defining a new root module

2. `embeds_many :foo, :"Elixir.FooBar"` - the module is hardcoded as an atom. may come from a macro that has already expanded the module

Reply:
Thanks I've updated the PR with these changes. Just for my understanding, would you be able to tell me if I am "getting it":

Is prefixing a module name with `Elixir.` is basically an accepted convention that it will create a root module? For instance if I didn't treat it specially, `Module.create` would have no issue creating nested module `Root.Elixir.MyModule`?

Reply:
It is more than a convention. It is a compiler "rule". For example, if you do:

    alias List, as: MyList

`MyList` will resolve to `List` but `Elixir.MyList` always resolves to `MyList`.

> For instance if I didn't treat it specially, Module.create would have no issue creating nested module Root.Elixir.MyModule?

Yes, you can still have Elixir in the middle of the module name (although it is probably a confusing idea to do so!).

Reply:
Thank you this has been really interesting/insightful.

Reply:
And thank you for the PR <3